app/src/main/java/com/example/firstandroid/MainActivity.kt

```diff
@@ -26,115 +22,31 @@ import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.unit.dp
 import androidx.lifecycle.lifecycleScope
-import com.example.firstandroid.BuildConfig.BuildConfig
-import com.example.firstandroid.Presentation.NewBathroom.NewBathroomViewModel
-import com.example.firstandroid.Presentation.UpdateBathroom.UpdateBathroomData
-import com.example.firstandroid.Presentation.UpdateBathroom.UpdateBathroomDialog
-import com.example.firstandroid.Presentation.UpdateBathroom.UpdateBathroomViewModel
-import com.example.firstandroid.Presentation.GetBathroom.BathroomData
-import com.example.firstandroid.Presentation.NewBathroom.AddBathroomDialog
-import com.example.firstandroid.data.networking.ApiService
+import com.example.firstandroid.presentation.add_bathroom.NewBathroomViewModel
+import com.example.firstandroid.presentation.update_bathroom.UpdateBathroomViewModel
+import com.example.firstandroid.presentation.get_bathroom.GetBathroomData
+import com.example.firstandroid.presentation.add_bathroom.AddBathroomDialog
+import com.example.firstandroid.presentation.get_bathroom.GetBathroomDialog
+import com.example.firstandroid.presentation.get_bathroom.GetBathroomViewModel
 import kotlinx.coroutines.launch
-import retrofit2.Retrofit
-import retrofit2.converter.gson.GsonConverterFactory
 import java.math.BigDecimal
 
```

Comment:
Much cleaner activity after moving all that state out. 

I do not think you need to call lifecycleScope.launch. You should be able to run setContent directly. Before yo uwere calling it to make the API call I believe.